packages/graphql-key-transformer/src/sync_vtl_remove/Query.syncComments.req.vtl

```diff
@@ -0,0 +1,164 @@
+  $util.qr($modelQueryExpression.expressionNames.put("#sortKey", $sk))
+  $util.qr($modelQueryExpression.expressionValues.put(":sortKey0", { "$SkType" : $ctx.args.get($sk).between[0] }))
+  $util.qr($modelQueryExpression.expressionValues.put(":sortKey1", { "$SkType": $ctx.args.get($sk).between[1] }))
+#end
+#if( !$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).eq) )
+  #set( $modelQueryExpression.expression = "$modelQueryExpression.expression AND #sortKey = :sortKey" )
+  $util.qr($modelQueryExpression.expressionNames.put("#sortKey", $sk))
+  $util.qr($modelQueryExpression.expressionValues.put(":sortKey", { "$SkType": $ctx.args.get($sk).eq }))
+#end
+#if( !$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).lt) )
+  #set( $modelQueryExpression.expression = "$modelQueryExpression.expression AND #sortKey < :sortKey" )
+  $util.qr($modelQueryExpression.expressionNames.put("#sortKey", $sk))
+  $util.qr($modelQueryExpression.expressionValues.put(":sortKey", { "$SkType": $ctx.args.get($sk).lt }))
+#end
+#if( !$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).le) )
+  #set( $modelQueryExpression.expression = "$modelQueryExpression.expression AND #sortKey <= :sortKey" )
+  $util.qr($modelQueryExpression.expressionNames.put("#sortKey", $sk))
+  $util.qr($modelQueryExpression.expressionValues.put(":sortKey", { "$SkType" : $ctx.args.get($sk).le }))
+#end
+#if( !$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).gt) )
+  #set( $modelQueryExpression.expression = "$modelQueryExpression.expression AND #sortKey > :sortKey" )
+  $util.qr($modelQueryExpression.expressionNames.put("#sortKey", $sk))
+  $util.qr($modelQueryExpression.expressionValues.put(":sortKey", { "$SkType" : $ctx.args.get($sk).gt }))
+#end
+#if( !$util.isNull($ctx.args.get($sk)) && !$util.isNull($ctx.args.get($sk).ge) )
+  #set( $modelQueryExpression.expression = "$modelQueryExpression.expression AND #sortKey >= :sortKey" )
+  $util.qr($modelQueryExpression.expressionNames.put("#sortKey", $sk))
+  $util.qr($modelQueryExpression.expressionValues.put(":sortKey", { "$SkType" : $ctx.args.get($sk).ge }))
+#end
+## [End] Applying Key Condition **
+## [Start]  Set query expression for @key **
+#if( !$scan )
+  #set( $limit = $util.defaultIfNull($context.args.limit, 100) )
+  #set( $QueryRequest = {
+  "version": "2018-05-29",
+  "operation": "Sync",
+  "limit": $limit,
+  "query": $modelQueryExpression
+} )
+  #if( !$util.isNull($ctx.args.sortDirection)
+                    && $ctx.args.sortDirection == "DESC" )
+    #set( $QueryRequest.scanIndexForward = false )
+  #else
+    #set( $QueryRequest.scanIndexForward = true )
+  #end
+#if( $context.args.nextToken ) #set( $QueryRequest.nextToken = $context.args.nextToken ) #end
+  #if( $filterMap != {} )
+    #set( $QueryRequest.filter = $util.parseJson($util.transform.toDynamoDBFilterExpression($filterMap)) )
+  #end
+  #if( $index != "dbTable" )
```

Comment:
Is this check used to see if there is at least one defined index one which to query on?

Reply:
yes and if there is no index defined , it will remove the index from VTL and directly query on table

Reply:
nit: could we check here if the value of index is nullorempty as opposed to a default value? Do we still need to add DbTable Key/Value pairs in the PkMap?

Reply:
I have to add some default value to make sure selected (pk,sk) tuple belongs to some table or GSI or they are just filter parameters. If the default value isn't present , Its difficult to tell whether its a query operation on table or just a scan on table